api/client/diff.go

```diff
@@ -25,13 +28,14 @@ func (cli *DockerCli) CmdDiff(args ...string) error {
 		return err
 	}
 
-	outs := engine.NewTable("", 0)
-	if _, err := outs.ReadListFrom(body); err != nil {
+	changes := []types.ContainerChange{}
+	if err := json.Unmarshal([]byte(body), &changes); err != nil {
```

Comment:
use `json.NewDecoder(body).Decode(&changes)` instead, you'll need to simply remove the `readBody` wrap.